### PR TITLE
Replace `unsugar_if` function with `is_if` function

### PR DESCRIPTION
**Pull request by flip1995**
cc https://github.com/rust-lang/rust-clippy/pull/4123#discussion_r296581719

changelog: none

r? @Manishearth

### CODE REVIEW

**Comment by Manishearth** [comment]
We should just remove it -- it was useful when if and if let used different ASTs, but it's not anymore, you can just match on the if.

File: clippy_lints/src/formatting.rs [hunk 1]
```diff
@@ -135,10 +134,10 @@ fn check_assign(cx: &EarlyContext<'_>, expr: &ast::Expr) {
 }
 
 /// Implementation of the `SUSPICIOUS_ELSE_FORMATTING` lint for weird `else`.
-fn check_else(cx: &EarlyContext<'_>, expr: &ast::Expr) {
+fn check_else(cx: &EarlyContext<'_>, expr: &Expr) {
     if_chain! {
-        if let Some((then, &Some(ref else_))) = unsugar_if(expr);
-        if is_block(else_) || unsugar_if(else_).is_some();
+        if let ExprKind::If(_, then, Some(else_)) = &expr.node;
+        if is_block(else_) || is_if(else_);
```
Comment (flip1995) [inline clippy_lints/src/formatting.rs]: There are multiple occurences of this function in a `||` or `&&` expression. Removing this function would make the code way more complicated IMO

Comment (Manishearth) [reply clippy_lints/src/formatting.rs]: ah

**Comment by Manishearth** [comment]
@bors r+

fair enough!

**Pull request merged**